### PR TITLE
Decommission verifyHtml(MainContent)WithRetry and merge it with verifyHtml(MainContent) #4876

### LINKED ISSUE #4876: Decommission verifyHtml(MainContent)WithRetry and merge it with verifyHtml(MainContent)
Rather than having `verifyHtml(MainContent)WithRetry` in random places. The retry will not be called if the verification is immediately successful so this will not slow down any test.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #4876

### CODE REVIEW

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 1]
```diff
@@ -735,20 +735,26 @@ private AppPage verifyHtml(By by, String filePath, boolean isWithRetry) throws I
...
+            // The check is done multiple times with waiting times in between to account for
+            // certain elements to finish loading (e.g ajax load, panel collapsing/expanding).
+            for (int i = 0; i < maxRetryCount; i++) {
+                if (i == maxRetryCount - 1) {
+                    // Last retry count: do one last attempt and if it still fails,
+                    // throw assertion error and show the differences
+                    HtmlHelper.assertSameHtml(expected, actual, isPart);
+                    break;
+                }
+                if (HtmlHelper.areSameHtml(expected, actual, isPart)) {
```
Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: this line looks the same as line 749; just wondering where the assertion error is thrown.

Comment (wkurniawan07) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: It's not the same. `assert` throws the error upon failure, `are` will just let it go.

Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: I am sorry haha, I misread the lines...

So does that mean that we only consider this as a "test" if it fails? Cos I think if it passes there would be no assert check...

Comment (wkurniawan07) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: Why would we need an assert check if it already passes? The flow is like this:
- Try 1, passes -> all is well
- Try 1, fails -> wait for 1 sec, get the actual page source again, go to try 2
- ...
- Final try, passes -> all is well
- Final try, fails -> throw assertion error and display diff as how we know it

Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: True, it is not a problem

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 2]
```diff
@@ -723,10 +723,10 @@ public void verifyTablePattern(int tableNum, int column, String patternString){
      * @return The page (for chaining method calls).
      */
     public AppPage verifyHtml(String filePath) throws IOException {
-        return verifyHtml(null, filePath, false);
+        return verifyHtml(null, filePath);
     }
 
-    private AppPage verifyHtml(By by, String filePath, boolean isWithRetry) throws IOException {
+    private AppPage verifyHtml(By by, String filePath) throws IOException {
```
Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: Unrelated but just wondering, why are we returning AppPage for this method?

Comment (wkurniawan07) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: I've been wondering as well. There's no chain method call after `verifyHtml*` (and it doesn't make sense to make one anyway).

Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: A firsttimers issue? :-P

**Pull request merged**